integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProviderIntegrationTest.java

```diff
@@ -132,7 +132,7 @@ public void shouldCreateNewPatient() throws Exception {
 		assertThat(patient, validResource());
 		
 		// try to get new patient
-		response = get("/Patient/" + patient.getId()).accept(FhirMediaTypes.JSON).go();
+		response = get("/Patient/" + patient.getIdElement().getIdPart()).accept(FhirMediaTypes.JSON).go();
```

Comment:
Is this change strictly-speaking necessary? I understand that `id` elements can be more complex so that `getId()` doesn't just return the `id`, but in this case that would actually indicate a bug in the code and so the test *should* fail.

Reply:
If you try to step in through a debugger, you would see that `patient.getId()` would have the value as "null" and there was a gap in the patient translator because of which it was returning a new instance. 

Generally speaking, `patient.getId()` has the value as `Patient/<uuid>`, whereas `patient.getIdElement()` has the value as `<uuid>`, hence in my opinion this change is necessary.

Reply:
I see. That makes sense.